pkg/common/cns-lib/vsphere/hostsystem.go

```diff
@@ -219,3 +219,28 @@ func GetSharedDatastoresForHosts(ctx context.Context, hosts []*HostSystem) ([]*D
 	}
 	return sharedDatastores, nil
 }
+
+// GetAllAccessibleDatastoresForHosts returns datastores accessible to hosts mentioned in the input parameter.
+func GetAllAccessibleDatastoresForHosts(ctx context.Context, hosts []*HostSystem) ([]*DatastoreInfo, error) {
+	log := logger.GetLogger(ctx)
+	var allAccessibleDatastores []*DatastoreInfo
+
+	for _, host := range hosts {
+		accessibleDatastores, err := host.GetAllAccessibleDatastores(ctx)
+		if err != nil {
+			log.Warnf("failed to fetch datastores from host %+v. Error: %+v",
+				host, err)
+			continue
+		}
+		if len(accessibleDatastores) == 0 {
+			log.Infof("No datastores found for host %+v", host)
```

Comment:
return error from here. Just in case if we have permission issue and driver can not find datastore on the host.